src/libraries/Native/Unix/System.Native/pal_networking.c

```diff
@@ -364,6 +351,8 @@ int32_t SystemNative_GetHostEntryForName(const uint8_t* address, HostEntry* entr
             }
         }
     }
+#else
+    (void)address;
```

Comment:
is this here to suppress warning about unused parameter in the else case?

Reply:
Yep. if `HAVE_GETIFADDRS` is not defined the C-compiler complains about unused parameters, hence this "trick" to avoid the warnings. Cf. https://github.com/dotnet/runtime/pull/34633#discussion_r405119626